### PR TITLE
Add some options to format subcommand

### PR DESCRIPTION
**Pull request by moznion**
This pull-request introduces some options to `flutter format` subcommand.

- `--dry-run`, `-n`
- `--set-exit-if-changes`
- `--machine`, `-m`

These options are passed through to `dartfmt` command.

This patch is related to #12797.

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/format.dart [hunk 1]
```diff
@@ -36,7 +56,24 @@ class FormatCommand extends FlutterCommand {
     }
 
     final String dartfmt = sdkBinaryName('dartfmt');
-    final List<String> cmd = <String>[dartfmt, '-w']..addAll(argResults.rest);
+    final List<String> cmd = <String>[dartfmt];
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/format.dart]: please rename "cmd" so that it's not an abbreviation (see our style guide). Thanks.

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/format.dart (hunk 1 above)
Comment (moznion) [reply packages/flutter_tools/lib/src/commands/format.dart]: Thank you for your pointed out. I fixed it at 3f1a299f2bc757423149726005fd173014906438.

**Comment by Hixie** [comment]
Thanks for your contribution! Sorry it took so long to land.

**Pull request merged**